src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -167,6 +184,32 @@ private AnalyzerExecutor(
             _addSuppression = addSuppression;
 
             _compilationAnalysisValueProviderFactory = new CompilationAnalysisValueProviderFactory();
+
+            if (getAnalyzerConfigOptionsProvider != null)
+            {
+                var hasDifferentOptions = false;
+
+                var map = new Dictionary<DiagnosticAnalyzer, AnalyzerOptions>(ReferenceEqualityComparer.Instance);
```

Comment:
Given this will have a large number of keys, is allocating this better than just calling the function each time and then just updating the map on 195 as you go?